VIC_RBM/model_run/result_analysis/scripts/plot_flow_future.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/local/anaconda/bin/python
+
+# This script plots different routed flow results together
+
+import numpy as np
+import datetime as dt
+import matplotlib.pyplot as plt
+import sys
+import pandas as pd
+import argparse
+import my_functions
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--USGS_code", type=str,  help="USGS code")
+parser.add_argument("--USGS_name", type=str,  help="USGS name")
+parser.add_argument("--hist_route_path", type=str, help="Historical route output path")
+parser.add_argument("--future_route_paths", nargs='+', type=str, help="eg. 'path1' 'path2'")
+parser.add_argument("--future_route_names", nargs='+', type=str, help="eg. 'Scenario1' 'Scenario2'")
+args = parser.parse_args()
+
+#========================================================
+# Parameter setting
+#========================================================
+usgs_site_code = args.USGS_code # '03571850'
+usgs_site_name = args.USGS_name # 'Tennessee River at Lower Pittsburgh'
+
+hist_route_path = args.hist_route_path # historical run. Lohmann routing output. year; month; day; flow(cfs)
+future_route_paths = args.future_route_paths # future run. Lohmann routing output. year; month; day; flow(cfs); ['path1', 'path2']
+future_route_names = args.future_route_names # future run. year; month; day; flow(cfs); ['Scenario1', 'Scenario2']
+
+output_plot_basename = '../output/Tennessee_' + usgs_site_code # output plot path basename (suffix will be added to different plots)
+
+#-------------------------------------------------
+
+plot_hist_start_date = dt.datetime(1951, 10, 1)  # start date shown on the plot for historical run (should be complete water years)
+plot_hist_end_date = dt.datetime(2010, 9, 30)  # end date shown on the plot for historical run
+
+plot_future_start_date = dt.datetime(1951, 10, 1)  # start date shown on the plot for future run (should be complete water years)
+plot_future_end_date = dt.datetime(2099, 9, 30)  # end date shown on the plot for future run
+
+time_locator = ('year', 20)  # time locator on the plot; 'year' for year; 'month' for month. e.g., ('month', 3) for plot one tick every 3 months
+
+#-------------------------------------------------
+
+nfuture = len(future_route_paths)  # number of future scenarios
+
+hist_style = 'k-'
+style_options = ['b--', 'r--', 'g--', 'm--', 'y--']
```

Comment:
I like this dynamic way of making a style list. But what happens if you have more than 5 future scenarios?